### PR TITLE
Galaxy 1948 unit test audit

### PR DESCRIPTION
**Pull request by hannahlspencer**

### CODE REVIEW

File: local-configuration/build.gradle [hunk 1]
```diff
@@ -27,4 +27,10 @@ dependencies {
     //Dagger
     implementation "com.google.dagger:dagger:${dagger_version}"
     kapt "com.google.dagger:dagger-compiler:${dagger_version}"
+
+    //Testing
+    testImplementation "junit:junit:${test_junit_version}"
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
```
Comment (TomMarshBBC) [inline local-configuration/build.gradle]: can we move this import outside of the testing bit pls

File: local-configuration/src/test/java/uk/co/bbc/nautilusapp/localconfiguration/SharedPreferencesRepositoryTest.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        on { getBoolean(context.getString(R.string.key_bottom_nav), false) }.thenReturn(true)
+        on { getBoolean(context.getString(R.string.key_dark_theme), false) }.thenReturn(true)
+        on { getBoolean(context.getString(R.string.key_autoplay_enabled), false) }.thenReturn(true)
+    }
+
+    private val listener = mock<SharedPreferences.OnSharedPreferenceChangeListener>()
+
+    private val sharedPreferencesRepository = SharedPreferencesRepository(context, sharedPreferences)
+
+    @Test
```
Comment (TomMarshBBC) [inline local-configuration/src/test/java/uk/co/bbc/nautilusapp/localconfiguration/SharedPreferencesRepositoryTest.kt]: These 3 tests here aren't really testing anything in our code, They are just testing the mocks.

We should just verify the code path by doing something like

`verify(sharedPreferences.getBoolean(context.getString(R.string.key_bottom_nav), false))`

for this to work you may have to mock out the context as well, for the `getString()`

File: ribbon/build.gradle [hunk 3]
```diff
@@ -2,6 +2,7 @@ apply plugin: 'com.android.dynamic-feature'
 apply plugin: 'kotlin-android'
 apply plugin: 'kotlin-android-extensions'
 apply plugin: 'kotlin-kapt'
+apply plugin: 'com.google.gms.google-services'
```
Comment (TomMarshBBC) [inline ribbon/build.gradle]: this should be at the bottom of the build gradle file. It gives you a compiler warning when it's at the top

File: settings/build.gradle [hunk 4]
```diff
@@ -28,6 +28,19 @@ dependencies {
     //Dagger
     implementation "com.google.dagger:dagger:${dagger_version}"
     kapt "com.google.dagger:dagger-compiler:${dagger_version}"
+
+    //Testing
+    testImplementation "junit:junit:${test_junit_version}"
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+    testImplementation "com.nhaarman:mockito-kotlin:${test_mockito_kotlin_version}"
+    testImplementation "org.mockito:mockito-inline:${test_mockito_inline_version}"
+    testImplementation 'androidx.test.espresso:espresso-core:3.2.0'
```
Comment (TomMarshBBC) [inline settings/build.gradle]: why do we have espresso and test runner imports here? kimchi should be providing them.

File: settings/src/main/java/bbc/co/uk/settings/PreferencesActivity.kt [hunk 5]
```diff
@@ -18,6 +20,14 @@ class PreferencesActivity : AppCompatActivity(), SharedPreferences.OnSharedPrefe
     @Inject
     lateinit var restartUseCase: RestartUseCase
 
+    companion object {
```
Comment (TomMarshBBC) [inline settings/src/main/java/bbc/co/uk/settings/PreferencesActivity.kt]: why do we need this?

File: settings/src/test/java/bbc/co/uk/settings/ExtensionsKtTest.kt [hunk 6]
```diff
@@ -0,0 +1,35 @@
+package bbc.co.uk.settings
+
+import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.app.AppCompatDelegate
+import com.nhaarman.mockito_kotlin.mock
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+class ExtensionsKtTest {
```
Comment (TomMarshBBC) [inline settings/src/test/java/bbc/co/uk/settings/ExtensionsKtTest.kt]: These are only really testing against the mocks, and not business logic. All this stuff is view logic anyway, so you can delete this file

File: settings/src/main/java/bbc/co/uk/settings/PreferencesActivity.kt (hunk 5 above)
Comment (hannahlspencer) [reply settings/src/main/java/bbc/co/uk/settings/PreferencesActivity.kt]: This was leftover from some preferences test experimentation, will remove!

**Pull request merged**